service-statistics-pxweb/src/main/java/fi/nls/oskari/control/statistics/plugins/pxweb/parser/ValueProcessor.java

```diff
@@ -0,0 +1,32 @@
+package fi.nls.oskari.control.statistics.plugins.pxweb.parser;
+
+import fi.nls.oskari.control.statistics.data.IndicatorValueFloat;
+import fi.nls.oskari.control.statistics.data.RegionValue;
+import fi.nls.oskari.control.statistics.plugins.db.DatasourceLayer;
+import org.json.JSONArray;
+
+import java.util.Optional;
+
+/**
+ * Returns the region ID/value pair if one is found.
+ */
+public class ValueProcessor {
```

Comment:
Maybe this could be an exciting JDK8 interface with a default method. e.g. `default public Optional<RegionValue> getRegionValue(JSONArray values, String regionId, int valueIndex, DatasourceLayer layer) { ... }`

Reply:
Scratch that, we can't instantiate that :D Maybe this should still implement an interface still, because that's what we want to do here.

Reply:
I'm not at all sure that this is an hook/plugin interface we are going to keep in the long run. The function params seem kinda weird to me. Anyway it fulfills the requirement for now and can be improved upon later.